src/libsyntax/parse/unescape.rs

```diff
@@ -66,6 +66,28 @@ where
     })
 }
 
+/// Takes a contents of a string literal (without quotes) and produces a
+/// sequence of characters or errors.
+/// NOTE: Raw strings do not perform any explicit character escaping, here we
+/// only translate CRLF to LF and produce errors on bare CR.
+pub(crate) fn unescape_raw_str<F>(literal_text: &str, callback: &mut F)
+where
+    F: FnMut(Range<usize>, Result<char, EscapeError>),
+{
+    let mut byte_offset: usize = 0;
+
+    let mut chars = literal_text.chars().peekable();
+    while let Some(curr) = chars.next() {
+        let result = match (curr, chars.peek()) {
+            ('\r', Some('\n')) => Ok(curr),
```

Comment:
Looks like some special handling is missing here:

* return `\n`
*  bump `chars` one more time
* make byte_offset.. range spanning two chars

Reply:
Ah, the description of PR says that this is a WIP :)

Reply:
If possible, I'd like to do this as a separate PR :)

Reply:
(hopefully correctly) done in https://github.com/rust-lang/rust/pull/60793/commits/d4632744fa0fb18a7c3f5058f1e8157c760353b4